flink-clients/src/main/java/org/apache/flink/client/cli/CancelOptions.java

```diff
@@ -33,11 +37,21 @@
     /** Optional target directory for the savepoint. Overwrites cluster default. */
     private final String targetDirectory;
 
+    private final SavepointFormatType formatType;
+
     public CancelOptions(CommandLine line) {
         super(line);
         this.args = line.getArgs();
         this.withSavepoint = line.hasOption(CANCEL_WITH_SAVEPOINT_OPTION.getOpt());
         this.targetDirectory = line.getOptionValue(CANCEL_WITH_SAVEPOINT_OPTION.getOpt());
+        if (line.hasOption(SAVEPOINT_FORMAT_OPTION)) {
+            formatType =
+                    ConfigurationUtils.convertValue(
+                            line.getOptionValue(SAVEPOINT_FORMAT_OPTION),
+                            SavepointFormatType.class);
+        } else {
+            formatType = SavepointFormatType.DEFAULT;
```

Comment:
Maybe is it better to do these `if-else` things inside `SavepointFormatType` and here just call `SavepointFormatType.fromString()`?

Reply:
Which ifs? We are not converting from a string here, but from a `CommandLine`. For that reason I don't think `SavepointFormatType` is the correct place, but I'll give it a second thought where in the cli package we could put that code.

Reply:
I am talking about this:
```
if (line.hasOption(SAVEPOINT_FORMAT_OPTION)) {
            formatType =
                    ConfigurationUtils.convertValue(
                            line.getOptionValue(SAVEPOINT_FORMAT_OPTION),
                            SavepointFormatType.class);
        } else {
            formatType = SavepointFormatType.DEFAULT;
}
```
I actually expect here something like this:
```
formatType = SavepointFormatType.fromString(line.getOptionValue(SAVEPOINT_FORMAT_OPTION));
```
Another proposal to create `convertValue` method with default:
```
ConfigurationUtils.convertValue(
                            line.getOptionValue(SAVEPOINT_FORMAT_OPTION),
                            SavepointFormatType.class, SavepointFormatType.DEFAULT);
```
But in general, I understand your point that it is a kind of different representation level. But anyway I would expect the method something like `getOrDefault(line, SAVEPOINT_FORMAT_OPTION)`. In fact, it is not critical for me, if you think that any changes don't make any sense we can leave it as is.

Reply:
I could do:
```
        this.formatType =
                ConfigurationUtils.convertValue(
                        line.getOptionValue(
                                SAVEPOINT_FORMAT_OPTION, SavepointFormatType.DEFAULT.toString()),
                        SavepointFormatType.class);
```

Do you think that would be better? I would not like to extend the `convertValue` to handle default values or treat `null` specially. The default value for `ConfigOption(s)` is handled on a higher level.

Reply:
I think you can leave it as is now. I still feel that it is not great since I see the pretty classic pattern `deserialize-if-not-null-or-default` but I am not ready to say what will be better since we already have several layers String->ConfigOption->String->Object(or something similar) and I don't want to make it more difficult.